opentracing-brave/src/main/java/io/opentracing/BraveSpanImpl.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.ServerTracer;
+import com.github.kristofa.brave.SpanId;
+
+
+final class BraveSpanImpl extends AbstractSpan {
+
+    final SpanId spanId;
+    final Optional<Span> parent;
+    final Optional<ServerTracer> serverTracer;
```

Comment:
sure. I was just looking at this impl vs the one in golang, for example.

What I noticed was that the act of injecting or extracting didn't seem to
have any side-effects on the span (like adding an "cs" annotation, for
example)
https://github.com/openzipkin/zipkin-go-opentracing/blob/master/propagation_ot.go

Most of the zipkin-specific logic ended up in a recorder (which calls a
collector incidentally very similar to brave's)
https://github.com/openzipkin/zipkin-go-opentracing

This is also similar to Uber's (eventhough Uber's doesn't use zipkin
propagation or transports).

Ex. the injector converts to their format (as opposed to B3), without
modifying the span.
https://github.com/uber/jaeger-client-java/blob/master/jaeger-jaxrs2/src/main/java/com/uber/jaeger/propagation/JerseyClientRequestInjector.java#L41

And the reporter converts the span to a zipkin struct (without side-effects
on the input)

https://github.com/uber/jaeger-client-java/blob/194476bdd7caa403db83ac83a1bc08e389d1217c/jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java#L83

In Uber's they use opentracing tags as indicators if something is
client/server etc (you'll notice conversions internal in their span type)
https://github.com/uber/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/Span.java

Long story short, I was mainly curious how we wanted to go. The most
opentracing way seems to be isolating zipkin specifics to the propagation
and recorder components, and expecting users to use opentracing tags like
"peer.ipv4" which we'd convert internally to zipkin state.

If still unclear I suppose you are right that writing code would be the
next step towards clarity.